mlir/include/mlir/Interfaces/ControlFlowInterfaces.td

```diff
@@ -317,7 +391,7 @@ def RegionBranchTerminatorOpInterface :
       /*defaultImplementation=*/[{
         ::mlir::Operation *op = $_op;
         ::llvm::cast<::mlir::RegionBranchOpInterface>(op->getParentOp())
-          .getSuccessorRegions(op->getParentRegion(), regions);
+          .getSuccessorRegions(::llvm::cast<::mlir::RegionBranchTerminatorOpInterface>(op), regions);
```

Comment:
```suggestion
          .getSuccessorRegions($_op, regions);
```
Not sure if this works, but nicer if it does

Reply:
It does not: that produces the concrete op, which is not implicitly convertible to the interface.